### ISSUE TITLE
Issues with Model Output and Request for Pretrained Weights

### ISSUE THREAD

**Issue by WUyinwei-hah** [opendatalab/LEGION#6]
Hi, thank you for your great work!

I trained the model using the provided scripts, but encountered the following issues:

The generated masks are completely black.

Some outputs contain repetitive or nonsensical sentences.

To ensure I didn't miss anything during training, could you kindly share the pretrained model weights used in your experiments? This would help me better reproduce the results and troubleshoot these issues.

Thanks in advance for your help!

**Comment by melodyincopenhagen** [comment]
Hi Yinwei,

Thanks for sharing your experience. I noticed you were able to train the model, although you encountered issues with black masks and repetitive sentences later on.

I'm currently facing a different problem trying to run the training script (scripts/loc_exp/train.py). I'm getting a ModuleNotFoundError because directories like dataset/caption_datasets/ seem to be missing from the repository, and the script fails during the import stage in dataset/dataset.py.

Since you were able to proceed with training, did you obtain the missing dataset files from another source, modify the code to bypass these imports, or use a specific setup/configuration?

Any insights would be greatly appreciated.

Thanks!

**Comment by WUyinwei-hah** [comment]
> Hi Yinwei,
> 
> Thanks for sharing your experience. I noticed you were able to train the model, although you encountered issues with black masks and repetitive sentences later on.
> 
> I'm currently facing a different problem trying to run the training script (scripts/loc_exp/train.py). I'm getting a ModuleNotFoundError because directories like dataset/caption_datasets/ seem to be missing from the repository, and the script fails during the import stage in dataset/dataset.py.
> 
> Since you were able to proceed with training, did you obtain the missing dataset files from another source, modify the code to bypass these imports, or use a specific setup/configuration?
> 
> Any insights would be greatly appreciated.
> 
> Thanks!

Hi, you can find them in GLaMM's repo (https://github.com/mbzuai-oryx/groundingLMM). But make sure not to override legion's original files in dataset folder.

**Comment by melodyincopenhagen** [comment]
Thank you so much for your help!

**Comment by fantacyfromsjtu** [comment]
> Hi, thank you for your great work!
> 
> I trained the model using the provided scripts, but encountered the following issues:
> 
> The generated masks are completely black.
> 
> Some outputs contain repetitive or nonsensical sentences.
> 
> To ensure I didn't miss anything during training, could you kindly share the pretrained model weights used in your experiments? This would help me better reproduce the results and troubleshoot these issues.
> 
> Thanks in advance for your help!

I have met the same problem. Have you solved it?

**Comment by WUyinwei-hah** [comment]
> > Hi, thank you for your great work!
> > I trained the model using the provided scripts, but encountered the following issues:
> > The generated masks are completely black.
> > Some outputs contain repetitive or nonsensical sentences.
> > To ensure I didn't miss anything during training, could you kindly share the pretrained model weights used in your experiments? This would help me better reproduce the results and troubleshoot these issues.
> > Thanks in advance for your help!
> 
> I have met the same problem. Have you solved it?

Unfortunately, not yet.

**Comment by hongchengp** [comment]
> 你好，感谢你的精彩作品！
> 
> 我用提供的脚本训练了模型，但遇到了以下问题：
> 
> 生成的遮罩完全是黑色的。
> 
> 有些输出包含重复或无意义的句子。
> 
> 为了确保我在训练过程中没有遗漏任何内容，能否分享一下你实验中使用的预训练模型权重？这样我能更好地复现结果并排查这些问题。
> 
> 提前感谢大家的帮助！

我也遇到了同样的问题，请问你解决了吗

**Comment by zhangruotian** [comment]
> > > Hi, thank you for your great work!
> > > I trained the model using the provided scripts, but encountered the following issues:
> > > The generated masks are completely black.
> > > Some outputs contain repetitive or nonsensical sentences.
> > > To ensure I didn't miss anything during training, could you kindly share the pretrained model weights used in your experiments? This would help me better reproduce the results and troubleshoot these issues.
> > > Thanks in advance for your help!
> > 
> > 
> > I have met the same problem. Have you solved it?
> 
> Unfortunately, not yet.

I can't reproduce the paper's results whether I use the weights provided by the repo or train them myself, and the visualization results are also poor. Any progresses from your side? Thanks a lot. @WUyinwei-hah
